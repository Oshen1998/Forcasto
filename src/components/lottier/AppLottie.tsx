import React from 'react';
import LottieView from 'lottie-react-native';

export type AppLottieProps = {
  source: string;
  height: number;
  width: number;
};

const AppLottie = (props: AppLottieProps) => {
  const {source, height, width} = props;
  return (
    <LottieView
      testID="lottie"
      source={source}
      autoPlay
      loop
      style={{height: height, width: width}}
    />
  );
};

export default AppLottie;
