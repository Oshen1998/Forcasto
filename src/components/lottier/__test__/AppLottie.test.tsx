import React from 'react';
import {render, screen} from '@testing-library/react-native';
import AppLottie from '../AppLottie';

jest.mock('lottie-react-native', () => jest.fn());

describe('AppLottie Component', () => {
  it('should renders correctly', () => {
    const mockSource = 'mock-source';
    render(<AppLottie source={mockSource} height={100} width={100} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
