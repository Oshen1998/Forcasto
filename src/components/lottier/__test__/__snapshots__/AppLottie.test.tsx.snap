// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLottie Component renders correctly 1`] = `null`;

exports[`AppLottie Component should renders correctly 1`] = `null`;
